src/components/MajorFilter.jsx

```diff
@@ -30,17 +31,15 @@ function MajorFilter() {
             variant="outlined"
             color="primary"
             key={major}
-            onDelete={() =>
-              setMajors((chips) => chips.filter((chip) => chip !== major))
-            }
+            onDelete={() => changeFn(majors.filter((m) => m !== major))}
             sx={{ mx: 1, mt: 1, color: '#000' }}
           />
         ))}
         <Autocomplete
           multiple
           open={isOpen}
           value={majors}
-          onChange={(e, val) => setMajors(val)}
+          onChange={(e, val) => changeFn(val)}
```

Comment:
I'm noticing that if I select something already selected it deselects it. Maybe `options` should exclude anything already selected? You should be able to use set operations for this.